File: week-10/day-4/Inheritence/Inheritence/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Inheritence
{
    class Student : Person
    {
        private int skipDays;

        public Student(string name, int age, Gender gender) : base(name, age, gender)
        {
            skipDays = 0;
        }

        public int SkipDays { get => skipDays; set => skipDays = value; }

        public override void GetGoal()
        {
            Console.WriteLine("My goal is: Learn C#!");
        }

        public void UseAllMethods()
        {
            GetGoal();
            Introduce();
        }
    }
}

File: week-04/day-3/src/cloneable/Student.java
package cloneable;

public class Student extends Person{
    String previousOrganization;
    int skippedDays;

    public Student() {
        this.previousOrganization = "The Dalles Wathonka High School";
        this.skippedDays = 0;
    }

    public Student(String name, int age, String gender, String previousOrganization) {
        super(name, age, gender);
        this.previousOrganization = previousOrganization;
        this.skippedDays = 0;
    }

    @Override
    public void getGoal() {
        System.out.println("My goal is: Be a junior software developer.");
    }

    @Override
    public void introduce() {
        System.out.println("Hi, I'm " + name + ", a " + age + " year old " + gender + " from " + previousOrganization + " who skipped " + skippedDays + " days from the course already.");
    }

    public void skipDays(int numberOfDays) {
        skippedDays += numberOfDays;
    }

    @Override
    public Student clone(){
        return new Student(this.name, this.age, this.gender, this.previousOrganization);
    }
}
